src/librustc/middle/reachable.rs

```diff
@@ -228,14 +223,8 @@ impl<'a, 'tcx> ReachableContext<'a, 'tcx> {
                 continue
             }
 
-            match self.tcx.map.find(search_item) {
-                Some(ref item) => self.propagate_node(item, search_item),
-                None if search_item == ast::CRATE_NODE_ID => {}
-                None => {
-                    self.tcx.sess.bug(&format!("found unmapped ID in worklist: \
-                                               {}",
-                                              search_item))
-                }
+            if let Some(ref item) = self.tcx.map.find(search_item) {
+                self.propagate_node(item, search_item);
```

Comment:
Hmm, I'm not sure.
It looks like fields don't have `Def`s, so we can't do something like
```
 tcx.def_map.borrow().get(&search_item) {
                        Some(def::DefField(..)) => ...
```
Maybe there's some other way?